enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,260 @@
+
+## Design Details
+
+### Open Questions
+
+1. What platform should it support? (just start with none) 
+2. Bootable installation artifact?, perhpas the installer should produce the liveCD?
+3. How will bootkube.sh know which disk to use for the installation? 
+4. Number of revisions for the control plane - do we want to make the bootstrap static pods as close as possible to the final ones?
+5. Do we want to rename etcd static pod in advance?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have
+CI coverage informing the release.
+An end-to-end job using the bootstrap-in-place installation flow,
+based on the [installer UPI CI](https://github.com/openshift/release/blob/master/ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml#L507),
+ running an appropriate subset of the standard OpenShift tests
+will be created and configured to block accepting release images
+unless it passes.
+
+That end-to-end job should also be run against pull requests for
+the installer and cluster-bootstrap.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. Initial proposal
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+
+- Maturity levels
+    - [`alpha`, `beta`, `stable` in upstream Kubernetes][maturity-levels]
+    - `Dev Preview`, `Tech Preview`, `GA` in OpenShift
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+#### Examples
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+1. The API will be unavailable from time to time during the installation.
+2. Coreos-installer cannot be used in the cloud environment.
+      
+## Alternatives
```

Comment:
We also talked about and rejected the idea of running the bootstrap on the real disk instead of the live ISO.